### PR TITLE
api: Add total_issued_requests to Upstream Locality and Endpoint Stats.

### PR DESCRIPTION
**Pull request by karthikbox**
Description:
total_issued_requests field tracks the count of requests issued since
the last report. This field will be used for global load balancing
decisions.

Envoy maintains a stats counter rq_total to track total
requests made. By latching this counter when load reporting period
begins, we are able to count the total requests issued in a load
reporting interval. This information is then used to populate the field
total_issued_requests.

Risk Level: Low
Testing: //test/integration:load_stats_reporter passes.

### CODE REVIEW

File: api/envoy/api/v2/endpoint/load_report.proto [hunk 1]
```diff
@@ -45,6 +45,11 @@ message UpstreamLocalityStats {
   // aggregated over all endpoints in the locality.
   uint64 total_error_requests = 4;
 
+  // The total number of requests that were issued by this Envoy since
```
Comment (htuch) [inline api/envoy/api/v2/endpoint/load_report.proto]: How does this relate to `total_requests`? Would be good to integrate with the above comments.

File: api/envoy/api/v2/endpoint/load_report.proto [hunk 2]
```diff
@@ -97,6 +102,10 @@ message UpstreamEndpointStats {
   //   - DataLoss
   uint64 total_error_requests = 4;
 
+  // The total number of requests that were issued to this endpoint
+  // since the last report.
+  uint64 total_issued_requests = 7;
```
Comment (htuch) [inline api/envoy/api/v2/endpoint/load_report.proto]: Ditto.

File: api/envoy/api/v2/endpoint/load_report.proto [hunk 3]
```diff
@@ -45,6 +45,11 @@ message UpstreamLocalityStats {
   // aggregated over all endpoints in the locality.
   uint64 total_error_requests = 4;
 
+  // The total number of requests that were issued by this Envoy since
+  // the last report. This information is aggregated over all the
+  // upstream endpoints in the locality.
+  uint64 total_issued_requests = 8;
```
Comment (htuch) [inline api/envoy/api/v2/endpoint/load_report.proto]: Probably warrants an entry in version_history.md.

File: api/envoy/api/v2/endpoint/load_report.proto (hunk 1 above)
Comment (vishalpowar) [reply api/envoy/api/v2/endpoint/load_report.proto]: the total_requests definition (sum of other total_*_requests) is incorrect above, as in_progress is a gauge.
Suggest removing the comment altogether.

Nit: for consistency it should be total_in_progress_requests .. but i guess we cant do anything about it.

File: api/envoy/api/v2/endpoint/load_report.proto (hunk 3 above)
Comment (karthikbox) [reply api/envoy/api/v2/endpoint/load_report.proto]: Updated. Please take a look.

File: api/envoy/api/v2/endpoint/load_report.proto (hunk 1 above)
Comment (htuch) [reply api/envoy/api/v2/endpoint/load_report.proto]: Yeah, I agree, the comment is no longer useful now that we have this field and has technical issues. +1 to remove.

Comment (karthikbox) [reply api/envoy/api/v2/endpoint/load_report.proto]: Done. Please take a look again!

**Comment by karthikbox** [comment]
Please take a look again.

**Pull request merged**